Sources_v3/StreamChatUI/ChatChannelList/ChatChannelView.swift

```diff
@@ -90,13 +89,13 @@ open class ChatChannelView<ExtraData: UIExtraDataTypes>: UIView {
     // MARK: - Public
     
     open func setupAppearance() {
-        channelNameLabel.numberOfLines = 1
+        channelNameLabel.numberOfLines = 0
         channelNameLabel.font = UIFontMetrics(forTextStyle: .subheadline)
             .scaledFont(for: .boldSystemFont(ofSize: UIFont.systemFontSize))
     }
     
     open func setupLayout() {
-        embedUsingSystemSpacing(container)
+        embed(container, insets: directionalLayoutMargins)
```

Comment:
Used `embed(container, insets: directionalLayoutMargins)` because `embedUsingSystemSpacing(container)` with negative multipliers doesn't seem to work. Looks like the multipliers get clamped to the `0...1` range.

Reply:
I think the container view should be added to the hierarchy outside of `setupLayout()`. If you override this without calling super, the container view should still be there in the correct position imo. Ideally. you'd just change the way the subviews are stacked there.

Reply:
It seems to me that the main stack view should be pinned directly to the edges and we should use something like (not tested):
```
container.preservesSuperviewLayoutMargins = true
container.isLayoutMarginsRelativeArrangement = true
```

This way, the views inside the container will respect the layout margins of the cell. If users don't want to respect this, they can easily turn it off and layout the views edge-to-edge

Reply:
updated